### PR TITLE
Add xcom push/pull to nav suggestions tasks

### PR DESCRIPTION
**Pull request by quiiver**
https://airflow.apache.org/docs/apache-airflow-providers-cncf-kubernetes/stable/operators.html#how-does-xcom-work
https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/xcoms.html

### CODE REVIEW

**Comment by quiiver** [comment]
https://github.com/mozilla-services/merino-py/pull/285

Allows us to output the public urls of the json file in the success email

File: dags/merino_jobs.py [hunk 1]
```diff
@@ -171,6 +174,9 @@ def merino_job(
         subject="Navigational Suggestions Domain Metadata job successful",
         html_content="""
```
Comment (ncloudioj) [inline dags/merino_jobs.py:175]:
```suggestion
        html_content=f"""
```

File: dags/merino_jobs.py [hunk 2]
```diff
@@ -171,6 +174,9 @@ def merino_job(
...
         Job completed. Download the new top picks json file on GCS.
+        
+        Prod Top Pick JSON file: {{ task_instance.xcom_pull('nav_suggestions_prepare_domain_metadata_prod')['top_pick_url']}}
```
Comment (ncloudioj) [inline dags/merino_jobs.py:178]:
```suggestion
        Prod Top Pick JSON file: { task_instance.xcom_pull("nav_suggestions_prepare_domain_metadata_prod")["top_pick_url"]}
```

**Review by ncloudioj — APPROVED** [review]
Looks good. r+wc

File: dags/merino_jobs.py (hunk 1 above)
Comment (quiiver) [reply dags/merino_jobs.py:175]: i committed and then reverted because i remembered that this is a jinja template and not a fstring :)

**Pull request merged**